src/Stratis.Bitcoin.IntegrationTests.Common/TestHelper.cs

```diff
@@ -33,20 +33,42 @@ public static void WaitLoop(Func<bool> act, string failureReason = "Unknown Reas
 
         public static bool AreNodesSynced(CoreNode node1, CoreNode node2)
         {
-            if (node1.FullNode.Chain.Tip.HashBlock != node2.FullNode.Chain.Tip.HashBlock) return false;
-            if (node1.FullNode.ChainBehaviorState.ConsensusTip.HashBlock != node2.FullNode.ChainBehaviorState.ConsensusTip.HashBlock) return false;
-            if (node1.FullNode.GetBlockStoreTip().HashBlock != node2.FullNode.GetBlockStoreTip().HashBlock) return false;
-            if (node1.FullNode.MempoolManager().InfoAll().Count != node2.FullNode.MempoolManager().InfoAll().Count) return false;
-            if (node1.FullNode.WalletManager().WalletTipHash != node2.FullNode.WalletManager().WalletTipHash) return false;
-            if (node1.CreateRPCClient().GetBestBlockHash() != node2.CreateRPCClient().GetBestBlockHash()) return false;
+            if (node1.FullNode.Chain.Tip.HashBlock != node2.FullNode.Chain.Tip.HashBlock)
+                return false;
+
+            if (node1.FullNode.ChainBehaviorState.ConsensusTip.HashBlock != node2.FullNode.ChainBehaviorState.ConsensusTip.HashBlock)
+                return false;
+
+            if (node1.FullNode.GetBlockStoreTip().HashBlock != node2.FullNode.GetBlockStoreTip().HashBlock)
+                return false;
+
+            if (node1.FullNode.MempoolManager().InfoAll().Count != node2.FullNode.MempoolManager().InfoAll().Count)
+                return false;
+
+            if ((node1.FullNode.WalletManager().Wallets.Count != 0) && (node2.FullNode.WalletManager().Wallets.Count != 0))
```

Comment:
you can do `node1.FullNode.WalletManager().ContainsWallets`

Reply:
OK will do :)